backend/src/apiserver/resource/resource_manager_util.go

```diff
@@ -266,3 +266,16 @@ func convertPipelineIdToDefaultPipelineVersion(pipelineSpec *api.PipelineSpec, r
 	})
 	return nil
 }
+
+// Override pipeline name parameter if this is a v2 compatible pipeline.
+func overrideV2PipelineName(wf *util.Workflow, name string, namespace string) {
+	var pipelineRef string
+	if namespace != "" {
+		pipelineRef = fmt.Sprintf("namespace/%s/pipeline/%s", namespace, name)
```

Comment:
How do we make sure the `name` parameter passing in this function is not the format `pipeline/some_name`? Based on the dsl change in this PR, the client side `pipeline_name` dsl.PipelineParam is: 'pipeline/{pipeline_name}'

Reply:
The name passed here comes from KFP API, so it doesn't have `pipeline/` prefix.

![image](https://user-images.githubusercontent.com/4957653/125717077-02793305-81e6-4bdd-a4fa-86525b7fc04b.png)